From 140d322fa8a112135ef15dc4b95ad3f14642fe3a Mon Sep 17 00:00:00 2001
From: Hyun Kwon <hyun.kwon@xilinx.com>
Date: Fri, 25 Jul 2014 16:44:57 -0700
Subject: [PATCH 110/456] drm: xilinx: plane: Remove the vdma reset call

This patch comes from:
  https://github.com/Xilinx/linux-xlnx.git

Resetting the VDMA IP shouldn't be needed in the client driver.
This introduces the direct dependency on the VDMA driver as well.
Thus, remove the VDMA reset call in the plane driver.

Signed-off-by: Hyun Kwon <hyunk@xilinx.com>
Acked-by: Laurent Pinchart <laurent.pinchart@ideasonboard.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
(cherry picked from commit 5c978b9fa1a29005bc5eac52e0c3c1a27c259fc3)
Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 drivers/gpu/drm/xilinx/xilinx_drm_plane.c |    6 ------
 1 files changed, 0 insertions(+), 6 deletions(-)

diff --git a/drivers/gpu/drm/xilinx/xilinx_drm_plane.c b/drivers/gpu/drm/xilinx/xilinx_drm_plane.c
index 1c1d20e..ad97fbb 100644
--- a/drivers/gpu/drm/xilinx/xilinx_drm_plane.c
+++ b/drivers/gpu/drm/xilinx/xilinx_drm_plane.c
@@ -20,7 +20,6 @@
 #include <drm/drm_fb_cma_helper.h>
 #include <drm/drm_gem_cma_helper.h>
 
-#include <linux/amba/xilinx_dma.h>
 #include <linux/device.h>
 #include <linux/dmaengine.h>
 #include <linux/of_dma.h>
@@ -115,7 +114,6 @@ void xilinx_drm_plane_dpms(struct drm_plane *base_plane, int dpms)
 {
 	struct xilinx_drm_plane *plane = to_xilinx_plane(base_plane);
 	struct xilinx_drm_plane_manager *manager = plane->manager;
-	struct xilinx_vdma_config dma_config;
 
 	DRM_DEBUG_KMS("plane->id: %d\n", plane->id);
 	DRM_DEBUG_KMS("dpms: %d -> %d\n", plane->dpms, dpms);
@@ -179,10 +177,6 @@ void xilinx_drm_plane_dpms(struct drm_plane *base_plane, int dpms)
 			xilinx_rgb2yuv_reset(plane->rgb2yuv);
 		}
 
-		/* reset vdma */
-		dma_config.reset = 1;
-		xilinx_vdma_channel_set_config(plane->dma.chan, &dma_config);
-
 		/* stop dma engine and release descriptors */
 		dmaengine_terminate_all(plane->dma.chan);
 		break;
-- 
1.7.5.4

